src/midje/parsing/0_to_fact_form/generative.clj

```diff
@@ -71,19 +73,86 @@
           num-tests               (or (:num-tests opts) 10)
           quick-check-opts        (->> (select-keys opts [:seed :max-size])
                                        (into [])
-                                       flatten)]
-      `(let [fact-fn#       (fn ~prop-names
+                                       flatten)
+          fact-fn-name            (gensym "fact-fn")]
+      `(let [~fact-fn-name       (fn ~prop-names
                               ~(parse-facts/wrap-fact-around-body
                                  metadata checks))
-             prop#          (prop/for-all* ~prop-values fact-fn#)
+             prop#          (prop/for-all* ~prop-values ~fact-fn-name)
              [run# passes#] (run-for-all (list ~num-tests
                                                prop#
                                                ~@quick-check-opts))]
+         (if (:result run#)
```

Comment:
this change reverts https://github.com/marick/Midje/pull/445/files#diff-fd516c514323a46d4fdd06b469bc8c4dL82
is this intentional?

Reply:
It wasn't intentional and I'm reverting. However, a strange thing is that the test added by that PR - https://github.com/marick/Midje/pull/445/files#diff-d356953a2908279065444c22223da180R128 - still passes with the bad (`:result`) code. Maybe something to take a better look at sometime.